RxSwift/Observables/Implementations/Materialize.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Materialize.swift
+//  Rx
+//
+//  Created by sergdort on 08/03/2017.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+public protocol EventType {
+    associatedtype E
+    func asEvent() -> Event<E>
+}
+
+extension Event : EventType {
+    public typealias E = Element
+    public func asEvent() -> Event<E> { return self }
+}
+
+fileprivate final class MaterializeSink<Element>: Disposable, ObserverType {
```

Comment:
Dont think we can use `Sink` here as it forwards `Event` but we want only forward `onNext`